### ISSUE TITLE
Allow use of Java 9

### ISSUE THREAD

**Issue by PaulMorris** [SolutionGuidance/psm#811]
Currently the PSM code base builds with Java 8, but not with Java 9.  Determine what needs to change to allow use of Java 9, and make those changes.

**Comment by PaulMorris** [comment]
Here's the error that occurs when attempting to build the PSM with OpenJDK 11 (the [default version](https://packages.ubuntu.com/bionic/default-jdk) under Ubuntu 18.04):
```
$ ./gradlew cms-portal-services:build

> Task :cms-business-model:xsd-dependency-tree
jaxb: starting Namespace Task
jaxb: generating xsd namespace dependency tree

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':cms-business-model:xjc'.
> java.lang.NoClassDefFoundError: javax/activation/DataSource
```

**Comment by mwolfthal** [comment]
I believe this is due to the fact that JAXB is no longer on the default module path of JDK 9. It can be added explicitly using the --add-modules java.xml.bind flag.

**Comment by kfogel** [comment]
Noting our goal: to have the PSM build and run with *both* Java 8 and Java 9 (and, if it's not much extra trouble, Java 10 too).

**Comment by mwolfthal** [comment]
JDK 9 was declared EOL in March 2018 so we should consider support only for JDK 10.
There are currently some issues open in gradle / groovy with JDK 10, especially as regards the use of reflection. We should wait for updates to these tools before proceeding with any adaptations.
We will probably need to begin by reviewing the module requirements of PSM as a whole to see whether some reorganization is advisable / required.
Initial testing indicates that Wildfly 12 runs on JDK 10.

**Comment by jasonaowen** [comment]
> There are currently some issues open in gradle / groovy with JDK 10, especially as regards the use of reflection. We should wait for updates to these tools before proceeding with any adaptations.

@mwolfthal what are those issues? Can you please put links to them here? That way we can know when we're unblocked.

> JDK 9 was declared EOL in March 2018 so we should consider support only for JDK 10.

Are the issues with gradle and groovy related to JDK 9, or JDK 10? It might be useful for us to fix the problem that blocked us from supporting JDK 9, so that the later upgrade to JDK 10 would be easier.

**Comment by mwolfthal** [comment]
I don't believe it is worthwhile to try to support JDK9 because packages that are deprecated in 9 are removed - or going to be - in 10, so it is hard to say how much of that work would remain valid.
On the groovy issue, see for example  https://issues.apache.org/jira/browse/GROOVY-8339:
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/C:/Program%20Files/gradle-4.2/lib/groovy-all-2.4.11.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
